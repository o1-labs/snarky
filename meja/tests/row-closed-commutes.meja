let f = fun (x1 : [> `A],
             x2 : [> `B],
             x3 : [< `A | `B | `C],
             x4 : [`A | `B | `D],
             b) => {
  let y1 = if (b) { x1; } else { x2; };
  let y2 = if (b) { x2; } else { x3; };
  let y3 = if (b) { x1; } else { x4; };
  (y1, y2, y3);
};
