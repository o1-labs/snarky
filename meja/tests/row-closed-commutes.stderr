File "tests/row-closed-commutes.meja", line 8, characters 33-35:
Error: Incompatible types ([<`A | `B | `C > `A | `B] as 'a) and
       [`A | `B | `D]:
       Cannot unify ([<`A | `B | `C > `A | `B] as 'a) and [`A | `B | `D]:
       the constructor `D is not present in both.
